### PR TITLE
The Joris Experience

### PR DESCRIPTION
**Pull request by Breadbox2k19**
## Additions

The most important animal mod ever made.

## Stuff
Individual animal tests have yet to be made. Currently, the creature known as Fat Jorry uses MVCF for its minigun, attempts to replace the vanilla shoot verb with CE shoot verb results in failure. If anyone else can make headway into the matter, please let me know.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long)

### CODE REVIEW

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Compatible, but at what cost?

Remarks:
- The Emo Joris needs its melee patched.
- The IsaacJoris "tear" weapon works fine, though I might either shorten the burst or decrease the ticksbetweenshots, since at the moment it's a rather time-consuming burst.
- The BiblicallyAccurateJoris and Joku have ranged attacks. While the attack seems to be working currently, they're horribly inaccurate, so they probably needs base accuracy assigned.
- Uncommenting the patch for the TorsoMinigun hediff and testing it, it seems to work just fine (you can tame the FatJoris and manually force it to shoot at stuff).

**Comment by Breadbox2k19** [comment]
> Compatible, but at what cost?
> 
> Remarks:
> 
> * The Emo Joris needs its melee patched.
> * The IsaacJoris "tear" weapon works fine, though I might either shorten the burst or decrease the ticksbetweenshots, since at the moment it's a rather time-consuming burst.
> * The BiblicallyAccurateJoris and Joku have ranged attacks. While the attack seems to be working currently, they're horribly inaccurate, so they probably needs base accuracy assigned.
> * Uncommenting the patch for the TorsoMinigun hediff and testing it, it seems to work just fine (you can tame the FatJoris and manually force it to shoot at stuff).

Impossible! this can't be real!

> Compatible, but at what cost?
> 
> Remarks:
> 
> * The Emo Joris needs its melee patched.
> * The IsaacJoris "tear" weapon works fine, though I might either shorten the burst or decrease the ticksbetweenshots, since at the moment it's a rather time-consuming burst.
> * The BiblicallyAccurateJoris and Joku have ranged attacks. While the attack seems to be working currently, they're horribly inaccurate, so they probably needs base accuracy assigned.
> * Uncommenting the patch for the TorsoMinigun hediff and testing it, it seems to work just fine (you can tame the FatJoris and manually force it to shoot at stuff).

Bible and joku joris has range attack because their range attack isn't actually patched at all.
They both use a custom shootverb and I'm not sure if they should be replaced by the CE verb without breakin something or changing what they are.

I'll take another look at fat jorry, tho I suspect it to be the same case.

**Comment by Breadbox2k19** [comment]
> Compatible, but at what cost?
> 
> Remarks:
> 
> * The Emo Joris needs its melee patched.
> * The IsaacJoris "tear" weapon works fine, though I might either shorten the burst or decrease the ticksbetweenshots, since at the moment it's a rather time-consuming burst.
> * The BiblicallyAccurateJoris and Joku have ranged attacks. While the attack seems to be working currently, they're horribly inaccurate, so they probably needs base accuracy assigned.
> * Uncommenting the patch for the TorsoMinigun hediff and testing it, it seems to work just fine (you can tame the FatJoris and manually force it to shoot at stuff).

Everything seems to work now.

File: Patches/The Joris Experience/JorisShootCE.xml [hunk 1]
```diff
@@ -44,6 +44,27 @@
 							</value>
 					</li>
 
+					<li Class="PatchOperationReplace">
+						<xpath>/Defs/ThingDef[defName="JE_Joku"]/verbs</xpath>
```
Comment (N7Huntsman) [inline Patches/The Joris Experience/JorisShootCE.xml:48]: This doesn't work. Honestly, I'd just leave the base attack in place and give both Joku and the BiblicalJoris better accuracy to make the vanilla attacks viable.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

File: Patches/The Joris Experience/JorisShootCE.xml (hunk 1 above)
Comment (Breadbox2k19) [reply Patches/The Joris Experience/JorisShootCE.xml:48]: Well, they are either aoe flamethrower attack or an orbital beam, not sure if accuracy benefit then much.

The range attack of bible joris is just weird, sometimes there's beam while other times there isn't.

Comment (N7Huntsman) [reply Patches/The Joris Experience/JorisShootCE.xml:48]: > Well, they are either aoe flamethrower attack or an orbital beam, not sure if accuracy benefit then much.

While they don't need to be pin-point accurate, they're currently inaccurate enough as for the creature to be often enough firing them in the the completely incorrect direction during testing. I've yet to successfully land a hit with the holy fire attack while testing it against a static pawn.

Comment (Breadbox2k19) [reply Patches/The Joris Experience/JorisShootCE.xml:48]: > While they don't need to be pin-point accurate, they're currently inaccurate enough as for the creature to be often enough firing them in the the completely incorrect direction during testing. I've yet to successfully land a hit with the holy fire attack while testing it against a static pawn.

CE seems to breaks JOKU and Biblejoris' attack even when unpatched, it's pissing me off, they work fine without CE but with CE it goes in the wrong direction without any error message. 
I can't patch bible joris' shootverb as its projectile an orbital beam.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**